asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
@willnorris Hmm, but if it's not exported API, how would @anpryl be able to achieve his original goal?

> It allows me to use single instance of my configured http.Client and set different Auth headers on each request.